src/Controls/tests/UITests/Tests/UITestBase.cs

```diff
@@ -98,5 +119,60 @@ public override TestConfig GetTestConfig()
 
 			return testConfig;
 		}
+
+		public void VerifyScreenshot(string? name = null)
+		{
+			if (UITestContext.TestConfig.TestDevice == TestDevice.Mac)
+			{
+				// For now, ignore visual tests on Mac Catalyst since the Appium screenshot on Mac (unlike Windows)
+				// is of the entire screen, not just the app. Later when xharness relay support is in place to
+				// send a message to the MAUI app to get the screenshot, we can use that to just screenshot
+				// the app.
+				Assert.Ignore("MacCatalyst isn't supported yet for visual tests");
+			}
+
+			if (name == null)
+				name = TestContext.CurrentContext.Test.MethodName;
+
+			IApp2? app = App as IApp2;
+			if (app is null)
+				throw new InvalidOperationException("App is not an IApp2");
+
+			byte[] screenshotPngBytes = app.Screenshot();
+			if (screenshotPngBytes is null)
+				throw new InvalidOperationException("Failed to get screenshot");
+
+			var actualImage = new ImageSnapshot(screenshotPngBytes, ImageSnapshotFormat.PNG);
+
+			// For Android and iOS, crop off the OS status bar at the top since it's not part of the
+			// app itself and contains the time, which always changes
+			int topStatusBarHeight = _testDevice switch
+			{
+				TestDevice.Android => 60,
+				TestDevice.iOS => 90,
+				_ => -1,
+			};
+
+			if (topStatusBarHeight != -1)
+			{
+				IImageEditor imageEditor = _imageEditorFactory.CreateImageEditor(actualImage);
+				(int width, int height) = imageEditor.GetSize();
+
+				imageEditor.Crop(0, topStatusBarHeight, width, height - topStatusBarHeight);
+
+				actualImage = imageEditor.GetUpdatedImage();
+			}
+
+			string platform = _testDevice switch
```

Comment:
Consider using this property instead of doing this work: UITestContext.TestConfig.PlatformName